code/datums/diseases/advance/advance.dm

```diff
@@ -385,7 +400,7 @@
 				if(!D.HasSymptom(S))
 					D.symptoms += S
 					i -= 1
-	while(i > 0)

```

Comment:
The do-while is best here.
@Cyberboss Not relevant, the input doesn't let you make symptom-less viruses anyway.